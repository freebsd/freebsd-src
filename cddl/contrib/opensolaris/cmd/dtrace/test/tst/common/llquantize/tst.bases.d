/*
 * CDDL HEADER START
 *
 * The contents of this file are subject to the terms of the
 * Common Development and Distribution License (the "License").
 * You may not use this file except in compliance with the License.
 *
 * You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
 * or http://www.opensolaris.org/os/licensing.
 * See the License for the specific language governing permissions
 * and limitations under the License.
 *
 * When distributing Covered Code, include this CDDL HEADER in each
 * file and include the License file at usr/src/OPENSOLARIS.LICENSE.
 * If applicable, add the following below this CDDL HEADER, with the
 * fields enclosed by brackets "[]" replaced with your own identifying
 * information: Portions Copyright [yyyy] [name of copyright owner]
 *
 * CDDL HEADER END
 */

/*
 * Copyright (c) 2011, Joyent, Inc. All rights reserved.
 */

#pragma D option quiet

tick-1ms
/i++ <= 100/
{
	@two = llquantize(i, 2, 0, 6, 2);
	@three = llquantize(i, 3, 0, 1, 9);
	@four = llquantize(i, 4, 0, 1, 4);
	@five = llquantize(i, 5, 0, 1, 25);
	@six = llquantize(i, 6, 0, 3, 12);
	@seven = llquantize(i, 7, 0, 1, 7);
	@eight = llquantize(i, 8, 0, 1, 16);
	@nine = llquantize(i, 9, 0, 1, 9);
	@ten = llquantize(i, 10, 0, 1, 10);
}

tick-1ms
/i > 100/
{
	exit(0);
}
