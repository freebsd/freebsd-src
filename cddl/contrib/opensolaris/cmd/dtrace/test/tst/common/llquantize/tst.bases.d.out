

           value  ------------- Distribution ------------- count    
             < 1 |                                         0        
               1 |                                         1        
               2 |@                                        2        
               4 |@@                                       4        
               8 |@@@                                      8        
              16 |@@@@@@                                   16       
              32 |@@@@@@@@@@@@@                            32       
              64 |@@@@@@@@@@@@@@@                          38       
          >= 128 |                                         0        


           value  ------------- Distribution ------------- count    
             < 1 |                                         0        
               1 |                                         1        
               2 |                                         1        
               3 |                                         1        
               4 |                                         1        
               5 |                                         1        
               6 |                                         1        
               7 |                                         1        
               8 |                                         1        
            >= 9 |@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@    93       


           value  ------------- Distribution ------------- count    
             < 1 |                                         0        
               1 |                                         1        
               2 |                                         1        
               3 |                                         1        
               4 |@@                                       4        
               8 |@@                                       4        
              12 |@@                                       4        
           >= 16 |@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@       86       


           value  ------------- Distribution ------------- count    
             < 1 |                                         0        
               1 |                                         1        
               2 |                                         1        
               3 |                                         1        
               4 |                                         1        
               5 |                                         1        
               6 |                                         1        
               7 |                                         1        
               8 |                                         1        
               9 |                                         1        
              10 |                                         1        
              11 |                                         1        
              12 |                                         1        
              13 |                                         1        
              14 |                                         1        
              15 |                                         1        
              16 |                                         1        
              17 |                                         1        
              18 |                                         1        
              19 |                                         1        
              20 |                                         1        
              21 |                                         1        
              22 |                                         1        
              23 |                                         1        
              24 |                                         1        
           >= 25 |@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@           77       


           value  ------------- Distribution ------------- count    
             < 1 |                                         0        
               1 |                                         1        
               2 |                                         1        
               3 |                                         1        
               4 |                                         1        
               5 |                                         1        
               6 |@                                        3        
               9 |@                                        3        
              12 |@                                        3        
              15 |@                                        3        
              18 |@                                        3        
              21 |@                                        3        
              24 |@                                        3        
              27 |@                                        3        
              30 |@                                        3        
              33 |@                                        3        
              36 |@@@@@@@                                  18       
              54 |@@@@@@@                                  18       
              72 |@@@@@@@                                  18       
              90 |@@@@@                                    12       
             108 |                                         0        


           value  ------------- Distribution ------------- count    
             < 1 |                                         0        
               1 |                                         1        
               2 |                                         1        
               3 |                                         1        
               4 |                                         1        
               5 |                                         1        
               6 |                                         1        
               7 |@@@                                      7        
              14 |@@@                                      7        
              21 |@@@                                      7        
              28 |@@@                                      7        
              35 |@@@                                      7        
              42 |@@@                                      7        
           >= 49 |@@@@@@@@@@@@@@@@@@@@@                    53       


           value  ------------- Distribution ------------- count    
             < 1 |                                         0        
               1 |                                         1        
               2 |                                         1        
               3 |                                         1        
               4 |                                         1        
               5 |                                         1        
               6 |                                         1        
               7 |                                         1        
               8 |@@                                       4        
              12 |@@                                       4        
              16 |@@                                       4        
              20 |@@                                       4        
              24 |@@                                       4        
              28 |@@                                       4        
              32 |@@                                       4        
              36 |@@                                       4        
              40 |@@                                       4        
              44 |@@                                       4        
              48 |@@                                       4        
              52 |@@                                       4        
              56 |@@                                       4        
              60 |@@                                       4        
           >= 64 |@@@@@@@@@@@@@@@                          38       


           value  ------------- Distribution ------------- count    
             < 1 |                                         0        
               1 |                                         1        
               2 |                                         1        
               3 |                                         1        
               4 |                                         1        
               5 |                                         1        
               6 |                                         1        
               7 |                                         1        
               8 |                                         1        
               9 |@@@@                                     9        
              18 |@@@@                                     9        
              27 |@@@@                                     9        
              36 |@@@@                                     9        
              45 |@@@@                                     9        
              54 |@@@@                                     9        
              63 |@@@@                                     9        
              72 |@@@@                                     9        
           >= 81 |@@@@@@@@                                 21       


           value  ------------- Distribution ------------- count    
             < 1 |                                         0        
               1 |                                         1        
               2 |                                         1        
               3 |                                         1        
               4 |                                         1        
               5 |                                         1        
               6 |                                         1        
               7 |                                         1        
               8 |                                         1        
               9 |                                         1        
              10 |@@@@                                     10       
              20 |@@@@                                     10       
              30 |@@@@                                     10       
              40 |@@@@                                     10       
              50 |@@@@                                     10       
              60 |@@@@                                     10       
              70 |@@@@                                     10       
              80 |@@@@                                     10       
              90 |@@@@                                     10       
          >= 100 |@                                        2        

